packages/elements/src/photon-medication-search/photon-medication-search-component.tsx

```diff
@@ -16,12 +16,34 @@ import {
   TreatmentOption
 } from '@photonhealth/sdk/dist/types';
 import { CatalogStore } from '../stores/catalog';
+import { DisableList } from '../photon-multirx-form/photon-prescribe-workflow';
 
 import { ApolloClient } from '@apollo/client';
 import gql from 'graphql-tag';
 
 // Utility Functions
 
+function isTreatmentDisabled(
+  treatmentId: string | undefined,
+  disableList: DisableList | undefined
+): { disabled: boolean; reason?: string } {
+  if (!treatmentId || !disableList || disableList.length === 0) {
+    return { disabled: false };
+  }
+
+  for (const item of disableList) {
+    if (item.treatmentIds) {
+      for (const disabledTreatmentId of item.treatmentIds) {
+        if (treatmentId === disabledTreatmentId) {
+          return { disabled: true, reason: item.reason };
+        }
+      }
+    }
+  }
```

Comment:
gonna nitpick the nested loop here. I can see a future way down the road where users abuse this with a long laundry list of blocking reasons, even against our own recommendation of usage. In this case the render time is `O(n*m*l)` where `n` is the number of reasons, `m` is the number of disabled treatments per reason, and `l` is the length of the medication list that we have to run this check against.

Try something like this: 
```suggestion
  const disableListsByReason = disableList.reduce((acc, cur) => {
    if (!cur.reason || !cur.treatmentIds) return acc;

    acc[cur.reason] = new Set(cur.treatmentIds);
    return acc;
  }, {} as Record<string, Set<string>>);

  for (const [reason, disabledTreatments] of Object.entries(disableListsByReason)) {
    if (disabledTreatments.has(treatmentId)) return { disabled: true, reason };
  }

```

Which would be `O(n*l)` at least. 

And if we instead move this check above the `displayTreatment` function so we only run the `reduce` and for loop once, then it's `O(n + l)`! Which, realistically will just be `O(l)` since the length of the medication list is likely longer than the length of reasons.

Reply:
yup that works for me